feat(builtin/k8s): Kubernetes improvements (#1317)
hashicorp/waypoint: .changelog/1317.txt

> ```release-note:feature

Author: Makes sense! Done!

```release-note:improvement

> plugins/k8s: Add more configuration options to Kubernetes plugin

Reviewer: ```suggestion
plugins/k8s: Add configuration options for pods and its security contexts
```
